src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -107,6 +106,11 @@ public static TOTPDevice registerDevice(AppIdentifierWithStorage appIdentifierWi
         }
 
         TOTPSQLStorage totpStorage = appIdentifierWithStorage.getTOTPStorage();
+        
+        if (deviceName == null) {
+            TOTPDevice[] devices = totpStorage.getDevices(appIdentifierWithStorage, userId);
+            deviceName = "TOTP Device " + (devices.length + 1);
```

Comment:
We need to only count the number of verified devices when auto creating a new device.

Reply:
Done.